hasktorch/test/Torch/Typed/FunctionalSpec.hs

```diff
@@ -985,6 +1011,16 @@ spec' device =
             hfoldrM @IO ModeSpec () (hproduct (hproduct dims keepOrDropDims) (hattach cuda0 (hproduct (withHalf standardDTypes) shapes)))
 
     describe "shape ops" $ do
+      it "narrow" $ do 
+        let dims = Proxy @0 :. Proxy @1 :. HNil
+            narrowStarts = Proxy @0 :. Proxy @1 :. HNil
+            narrowLengths = Proxy @1 :. Proxy @2 :. HNil
+            narrowShapes = Proxy @'[3, 3,2] :. Proxy @'[3, 3,2] :. HNil
+        case device of
+            D.Device { D.deviceType = D.CPU,  D.deviceIndex = 0 } ->
+              hfoldrM @IO NarrowSpec () (hproduct dims (hproduct narrowStarts (hproduct narrowLengths (hattach cpu   (hproduct standardFloatingPointDTypes narrowShapes)))))
+            D.Device { D.deviceType = D.CUDA, D.deviceIndex = 0 } ->
+              hfoldrM @IO NarrowSpec () (hproduct dims (hproduct narrowStarts (hproduct narrowLengths (hattach cuda0 (hproduct allFloatingPointDTypes      narrowShapes)))))
```

Comment:
```suggestion
            D.Device { D.deviceType = D.CPU,  D.deviceIndex = 0 } ->
              hfoldrM @IO NarrowSpec () (hproduct dims (hproduct narrowStarts (hproduct narrowLengths (hattach cpu   (hproduct allDTypes narrowShapes)))))
            D.Device { D.deviceType = D.CUDA, D.deviceIndex = 0 } ->
              hfoldrM @IO NarrowSpec () (hproduct dims (hproduct narrowStarts (hproduct narrowLengths (hattach cuda0 (hproduct allDTypes narrowShapes)))))
```
so that all `DType`s are checked :)